src/services/telemetryService.ts

```diff
@@ -0,0 +1,131 @@
+    clientExtensionName: string
+  ): string | undefined {
+    try {
+      const clientPackageJson = vscode.extensions.getExtension(
+        clientExtensionName
+      )?.packageJSON;
+      const clientSegmentKey = clientPackageJson['segmentWriteKey'];
+      Logger.log(`client Segment-Key : ${clientSegmentKey}`);
+      return clientSegmentKey;
+    } catch (error) {
+      Logger.log('Unable to get Client Segment-Key');
+    }
+    return undefined;
+  }
+
+  /* 
+    Collects telemetry data and pushes to a queue when not opted in
+    and to segment when user has opted for telemetry 
+  */
+  export function sendEvent(event: TelemetryEvent) {
+    Logger.log('Event received:');
+    Logger.log(event.uuid);
+    if (getTelemetryEnabledConfig()) {
+      if (analyticsObject) {
+        // setting analyticsObject for reporting
+        Reporter.setAnalytics(analyticsObject);
+        // dequeue the existing queue
+        // tslint:disable-next-line: no-unused-expression
+        TelemetryEventQueue.getQueue()?.length &&
+          Reporter.reportQueue(TelemetryEventQueue.getQueue());
+        // report event to segment
+        Reporter.report(event);
+      } else {
+        Logger.log('analytics was not initialized in vscode-commons');
+      }
+    } else {
+      TelemetryEventQueue.addEvent(event);
+    }
+  }
+
+  /* 
+      Checks the current telemetry configuration(package.json) 
+      returns "true if telemetry is enabled 
+      returns "false" if it is not-enable or undefined(default value)"
+  */
+  export function getTelemetryEnabledConfig(): boolean {
+    return vscode.workspace
+      .getConfiguration('redhat.telemetry')
+      .get<boolean>('enabled', false);
+  }
+
+  export function openTelemetryOptInDialogIfNeeded(
+    context: vscode.ExtensionContext
+  ) {
+    const optInRequested: boolean | undefined = context.globalState.get(
+      OPT_IN_STATUS
+    );
+    Logger.log(`optInRequested is: ${optInRequested}`);
+
+    if (!optInRequested) {
+      const privacyUrl: string =
+        'https://developers.redhat.com/article/tool-data-collection';
+      const optOutUrl: string =
+        'https://github.com/redhat-developer/vscode-commons/wiki/Usage-reporting';
+      const command: string = 'vscodeCommons.openWebPage';
+      const message: string = `Help Red Hat improve its extensions by allowing them to collect usage data. 
+                              Read our [privacy statement](command:${command}?"${privacyUrl}") 
+                              and learn how to [opt out](command:${command}?"${optOutUrl}").`;
+
+      vscode.window
+        .showInformationMessage(message, 'Accept', 'Deny')
+        .then((selection) => {
+          if (!selection) {
+            //close was chosen. Ask next time.
+            return;
+          }
+
+          context.globalState.update(OPT_IN_STATUS, true);
+
+          let optIn: boolean = selection === 'Accept';
+          updateTelemetryEnabledConfig(optIn);
+        });
```

Comment:
You can simply this using `await`.
```suggestion
      const selection = await vscode.window
        .showInformationMessage(message, 'Accept', 'Deny');
        if (!selection) {
            //close was chosen. Ask next time.
            return;
          }

          context.globalState.update(OPT_IN_STATUS, true);

          let optIn: boolean = selection === 'Accept';
          updateTelemetryEnabledConfig(optIn);
```

Reply:
sure :+1: